cdap-common/src/main/java/co/cask/cdap/common/conf/CConfigurationUtil.java

```diff
@@ -37,6 +41,29 @@ public static void copyTxProperties(CConfiguration cConf, org.apache.hadoop.conf
   }
 
   /**
+   * Get extra jars set in {@link CConfiguration} as a list of {@link URI}.
+   *
+   * @param cConf {@link CConfiguration} containing the extra jars
+   * @return a list of {@link File} created from the extra jars set in cConf.
+   */
+  public static List<URI> getExtraJars(CConfiguration cConf) {
+    String[] extraJars = cConf.getStrings(Constants.AppFabric.PROGRAM_CONTAINER_DIST_JARS);
+    List<URI> jarURIs = new ArrayList<>();
+    if (extraJars == null) {
+      return jarURIs;
+    }
+    for (String jarPath : extraJars) {
+      URI uri = URI.create(jarPath);
```

Comment:
So this will throw exception if `jarPath` is not an URI, right? Don't you need to handle that? (e.g. on Windows, it might have `\`. Also, user may have typo.